internal/controller/scheduler/job_watcher.go

```diff
@@ -58,7 +58,7 @@ func NewJobWatcher(logger *zap.Logger, k8sClient kubernetes.Interface, cfg *conf
 		logger:       logger,
 		k8s:          k8sClient,
 		cfg:          cfg,
-		stallingJobs: make(map[*batchv1.Job]struct{}),
+		stallingJobs: make(map[uuid.UUID]*batchv1.Job),
```

Comment:
Is there a mutex for stalling jobs or is covered by another?

Reply:
Yes, `stallingJobsMu` on line 40.